packages/mobx/src/types/observableset.ts

```diff
@@ -243,6 +243,58 @@ export class ObservableSet<T = any> implements Set<T>, IInterceptable<ISetWillCh
         } as any)
     }
 
+    intersection<U>(otherSet: ReadonlySetLike<U> & Partial<Set<U>>): Set<T & U> {
```

Comment:
Why the `& Partial<Set<U>>` ?

https://github.com/microsoft/TypeScript/blob/main/src/lib/esnext.collection.d.ts#L28-L57

Reply:
I see. Change it to `otherSet: ReadonlySetLike<U> | Set<U>` and the check to:
https://github.com/mobxjs/mobx/blob/5c8066d0ff6552bd15ab069a655c872b7da3395e/packages/mobx/src/utils/utils.ts#L154

Or are there any problems with that?

Reply:
Changed type to `| Set<>` and if branch check to `isES6Set`. There is no problems